### PR TITLE
Remove dep build directory from git history

### PR DESCRIPTION
**Pull request by ejguan**
For ubuntu and mac build.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,5 +12,14 @@ build/
 # OSX dir files
 .DS_Store
 
+# Dep
+_build/
+double-conversion/
```
Comment (wenleix) [inline .gitignore:17]: hmm... we should also fix the script so they don't pollute root level ? lol

**Review by wenleix — APPROVED** [review]
Thanks!

File: .gitignore (hunk 1 above)
Comment (wenleix) [reply .gitignore:17]: We can use this `DEPENDENCY_DIR` , defined in https://github.com/facebookresearch/torcharrow/blob/e03beb1750e13f96f8b8a049b6b82fb38c927841/scripts/_setup-macos.sh#L38

Here is an example usage in Velox CI: 
https://github.com/facebookincubator/velox/blob/776f9840462f7d2b5053740ffd61c5202eeeee67/.circleci/config.yml#L69

Comment (ejguan) [reply .gitignore:17]: I see. I can move them into `_build` directory.

Comment (ejguan) [reply .gitignore:17]: to align with the behavior on ubuntu

Comment (wenleix) [reply .gitignore:17]: @ejguan : Feel free to do it in a separate PR

**Pull request merged**